plugins/fcp_can_c/fcp_can_c/can_c_writer.py

```diff
@@ -366,10 +372,60 @@ def generate_device_sources(self) -> Generator[Tuple[str, str], None, None]:
         """
         for device_name, messages in self.device_messages.items():
             yield (
-                pascal_to_snake(device_name),
+                to_snake_case(device_name),
                 self.templates["device_can_c"].render(
-                    device_name_pascal=snake_to_pascal(device_name),
-                    device_name_snake=pascal_to_snake(device_name),
+                    device_name_pascal=to_pascal_case(device_name),
+                    device_name_snake=to_snake_case(device_name),
+                    messages=messages,
+                ),
+            )
+
+    def generate_rpc_headers(self) -> Generator[Tuple[str, str], None, None]:
+        """Generate C header files for devices with RPC.
+
+        Returns:
+            Generator: Tuple containing the device name and the file content.
+
+        """
+        for device in self.devices:
+            device_name = device.name
+            messages = self.device_messages.get(device_name, [])
+            rpc_get_id = device.rpc_get_id if device.rpc_get_id is not None else 0
+            rpc_ans_id = device.rpc_ans_id if device.rpc_ans_id is not None else 0
```

Comment:
Missing ids shouldn't default to anything. If the  device has no services, generating the device header can be skipped (I guess). If it has services then you should fail and provide an error message.